src/main/java/net/minestom/arena/feature/CombatFeature.java

```diff
@@ -60,29 +58,61 @@ public void hook(@NotNull EventNode<InstanceEvent> node) {
             if (!(event.getTarget() instanceof LivingEntity target)) return;
             if (!(event.getEntity() instanceof EntityProjectile projectile)) return;
 
+            // Don't apply damage if entity is invulnerable
+            final long now = System.currentTimeMillis();
+            final long invulnerableUntil = target.getTag(INVULNERABLE_UNTIL_TAG);
+            if (invulnerableUntil > now) return;
+
             float damage = (float) damageFunction.applyAsDouble(projectile, target);
 
             target.damage(DamageType.fromProjectile(projectile.getShooter(), projectile), damage);
+            target.setTag(INVULNERABLE_UNTIL_TAG, now + invulnerabilityFunction.applyAsLong(target));
```

Comment:
Storing such tag on entities themselves could cause issue if we decide to serialize (persistently) this tag handler

Reply:
I could use a Map<Player,TagHandler> for CombatFeature like we're doing for MobArena right now.

We could also decide to serialize another tag handler for persistency (would make more sense because anyone who takes a quick look at the source code won't know player tags are getting serialized)

Reply:
Hm maybe you are right. In this case we would also need to remove the current maps from MobArena.